contracts/vaults/PlasmaVault.sol

```diff
@@ -753,7 +755,55 @@ contract PlasmaVault is
 
         _addPerformanceFee(totalAssetsBefore);
 
-        return super.withdraw(assets_, receiver_, owner_);
+        uint256 maxAssets = maxWithdraw(owner_);
+
+        if (assets_ > maxAssets) {
+            revert ERC4626ExceededMaxWithdraw(owner_, assets_, maxAssets);
+        }
+
+        address withdrawManager = PlasmaVaultStorageLib.getWithdrawManager().manager;
+
+        uint256 shares = convertToShares(assets_);
+
+        if (withdrawManager != address(0)) {
+            uint256 feeSharesToBurn = WithdrawManager(withdrawManager).canWithdrawFromUnallocated(shares);
+            if (feeSharesToBurn > 0) {
+                uint256 assetsToWithdraw = assets_ - super.convertToAssets(feeSharesToBurn);
+
+                super._withdraw(_msgSender(), receiver_, owner_, assetsToWithdraw, shares - feeSharesToBurn);
+                _burn(owner_, feeSharesToBurn);
+                return assetsToWithdraw;
+            }
+        }
+
+        super._withdraw(_msgSender(), receiver_, owner_, assets_, shares);
+        return assets_;
+    }
+
+    function previewRedeem(uint256 shares_) public view override returns (uint256) {
```

Comment:
Add test which checks that previewRedeem with withdraw fee returns lover value of assets than previewRedeem without fee